### PR TITLE
Standardize DD_TAGS format

### PR DESCRIPTION
**Pull request by NouemanKHAL**
### Requirements for Contributing to this repository

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix one issue, or add one feature, at the time.

### What does this PR do?

This PR aims to add a standard interface to use the environment variable`DD_TAGS` in the buildpack regardless of the target language. As a result, this PR fixes all issues related to `DD_TAGS` format.

### Description of the Change

- Add support for custom `DD_TAGS` separator via `DD_TAGS_SEP`
- Pass custom tags to the tracing libraries via a comma-separated `DD_TAGS`
- Pass custom tags to the agents via their yaml configuration file (`tags`, `dogstatsd_tags`)
- Fix a typo in `DD_NODE_AGENT_TAGS` env variable
- Fix tags parsing bugs

### Review checklist (to be filled by reviewers)

- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: bin/supply [hunk 1]
```diff
@@ -57,18 +59,16 @@ chmod +x "${BUILD_DIR}/.profile.d/00-test-endpoint.sh"
...
 python "${DATADOG_DIR}/scripts/parse_env_vars.py" "${DATADOG_DIR}/.raw_datadog_env" "${DATADOG_DIR}/.datadog_env"
 
-# export DD_TAGS for ddtrace
-DD_TAGS=$(python "${DATADOG_DIR}"/scripts/get_tags.py)
-export DD_TAGS
-DD_DOGSTATSD_TAGS=$(python "${DATADOG_DIR}"/scripts/get_tags.py)
-export DD_DOGSTATSD_TAGS
+export DD_TAGS=$(python "${DATADOG_DIR}/scripts/get_tags.py")
 
 # mark the script as finished, useful to sync the update_agent_config script
```
Comment (sarah-witt) [inline bin/supply:73]: I think supply and compile should have the same content

File: lib/run-datadog.sh [hunk 2]
```diff
@@ -154,27 +146,27 @@ start_datadog() {
...
-        ./dogstatsd start --cfgpath dist/ &
+        env -u DD_TAGS ./dogstatsd start --cfgpath dist/ &
       fi
       echo $! > run/dogstatsd.pid
     fi
     if [ "${FIRST_RUN}" = "true" ]; then
       echo "Starting trace agent"
       if [ "${SUPPRESS_DD_AGENT_OUTPUT}" = "true" ]; then
-        ./trace-agent --config dist/datadog.yaml --pid run/trace-agent.pid > /dev/null 2>&1 &
+        env -u DD_TAGS ./trace-agent --config dist/datadog.yaml --pid run/trace-agent.pid > /dev/null 2>&1 &
```
Comment (sarah-witt) [inline lib/run-datadog.sh:167]: nice!

File: lib/scripts/update_datadog_config.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+end
+
+def get_tags()
+  dd_tags = File.file?(DD_TAGS_FILE) ? File.read(DD_TAGS_FILE) : nil
+  dd_node_agent_tags = ENV['DD_NODE_AGENT_TAGS'] || (File.file?(NODE_AGENT_TAGS_FILE) ? File.read(NODE_AGENT_TAGS_FILE) : nil)
+
+  tags = []
+
+  if !dd_tags.nil?
+    tags += sanitize(dd_tags, ",")
```
Comment (sarah-witt) [inline lib/scripts/update_datadog_config.rb:35]: This delimiter is not guaranteed to be a comma as it is coming from the user input

Comment (NouemanKHAL) [reply lib/scripts/update_datadog_config.rb:35]: It actually is, the `.dd_tags.txt` file contains the output of the `get_tags.py` script (always comma-separated)


```bash
# lib/run-datadog.sh
export DD_TAGS=$(python "${DATADOG_DIR}/scripts/get_tags.py")
echo "${DD_TAGS}" > "${DATADOG_DIR}/.dd_tags.txt"
```

Comment (sarah-witt) [reply lib/scripts/update_datadog_config.rb:35]: Oh nice, that's good!

File: bin/supply (hunk 1 above)
Comment (NouemanKHAL) [reply bin/supply:73]: Done!

File: README.md [hunk 4]
```diff
@@ -37,6 +37,19 @@ env:
   DD_API_KEY: <DATADOG_API_KEY>
 ```
 
+#### Assigning Tags
+
+Custom tags can be configured via the environmnet variable `DD_TAGS` in the application manifest.
```
Comment (maycmlee) [inline README.md:42]:
```suggestion
Custom tags can be configured with the environment variable `DD_TAGS` in the application manifest.
```

File: README.md [hunk 5]
```diff
@@ -37,6 +37,19 @@ env:
...
+
+The tags must be a comma separated list. To use a different separator, set the `DD_TAGS_SEP` variable to the desired separator.
```
Comment (maycmlee) [inline README.md:44]:
```suggestion
The tags must be a comma separated list. To use a different separator, set `DD_TAGS_SEP` to the desired separator.
```

**Review by maycmlee — COMMENTED** [review]
Just a couple of suggestions

**Comment by NouemanKHAL** [comment]
Thank you @maycmlee !

**Review by maycmlee — DISMISSED** [review]
👍  for docs

File: README.md [hunk 6]
```diff
@@ -37,6 +37,19 @@ env:
...
+
+Custom tags can be configured with the environment variable `DD_TAGS` in the application manifest.
+
+The tags must be a comma separated list. To use a different separator, set `DD_TAGS_SEP` to the desired separator.
+
+```yaml
+env:
+  DD_TAGS: "key1:value1,key2:value2,key3:value3"
+  DD_TAGS_SEP: "," 
+```
```
Comment (sarah-witt) [inline README.md:50]: Can we also include a note about where the custom tags are included?

**Review by sarah-witt — APPROVED** [review]

**Review by maycmlee — COMMENTED** [review]
Just a couple of edits for the note.

**Review by maycmlee — DISMISSED** [review]

File: lib/scripts/get_tags.rb [hunk 7]
```diff
@@ -4,16 +4,19 @@
...
-# if DD_TAGS[0] is comma or space, then set is as delimiter
-# else continue as usual
+NODE_AGENT_TAGS_FILE = "/home/vcap/app/.datadog/node_agent_tags.txt"
 
 def parse_tags(tags)
-  delimiter = ','
-  delimiter = ' ' if tags.count(' ') > tags.count(',')
   begin
+    delimiter = ','
+    delimiter = ' ' if tags.count(' ') > tags.count(',')
```
Comment (NouemanKHAL) [inline lib/scripts/get_tags.rb:12]: @sarah-witt keeping this line will keep this change backwards-compatible

**Review by sarah-witt — DISMISSED** [review]

**Review by sarah-witt — APPROVED** [review]

**Pull request merged**